node/bindnode/example_test.go

```diff
@@ -80,3 +82,104 @@ func ExamplePrototype_onlySchema() {
 	// Output:
 	// {"Friends":["Sarah","Alex"],"Name":"Michael"}
 }
+
+func ExampleWrap_withSchemaUsingStringjoinStruct() {
+	ts := schema.TypeSystem{}
+	ts.Init()
+	ts.Accumulate(schema.SpawnString("String"))
+	ts.Accumulate(schema.SpawnStruct("FooBarBaz",
+		[]schema.StructField{
+			schema.SpawnStructField("foo", "String", false, false),
+			schema.SpawnStructField("bar", "String", false, false),
+			schema.SpawnStructField("baz", "String", false, false),
+		},
+		schema.SpawnStructRepresentationStringjoin(":"),
+	))
+	schemaType := ts.TypeByName("FooBarBaz")
+
+	type FooBarBaz struct {
```

Comment:
How is this mapping working? Does it just use field order to figure out what to assign where? So the schema could have any arbitrary name but the values get assigned to whatever order is there?

Reply:
Yes, when mapping a provided IPLD schema with a provided Go type, it just cares about the "shape" matching, not any names. For a struct, that means that the fields should have matching types in the same order.

Doing the mapping via names or struct field tags would also be an option, though it would require more code and features, so I initially just went for the simplest approach from my point of view.

I haven't documented this, mind you, because I'm not 100% certain that this default behavior is right. I think most users would expect case-insensitive-matching by field name, similar to most other Go libraries. I'm not bothered by this example being public, but the semantics in this case might change.